src/test_l1_dsFPD.c

```diff
@@ -1154,20 +1305,22 @@ void test_l1_dsFPD_negative_dsSetFPColor (void)
 
     // Step 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
     // Step 04: Call dsSetFPColor() with all valid indicators but invalid colors
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    for (int i = 0; i < count; i++)
     {
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-        dsFPDColorConfig_t *colors = kIndicators[i].supportedColors;
+        numOfSupportedColors = UT_KVP_PROFILE_GET_LIST_COUNT("dsFPD/SupportedFPDIndicators/%d/supportedColors");
```

Comment:
no indexing here as well.  %d is not associated with any indexing variable or value.

Reply:
Done